Update `config` and Gradle. Add missing Readme instructions

yuri-sergiichuk opened the pull request:
In this PR I have updated the `config` sub-module to reference the latest changes. Also, I updated the Gradle to the latest v6.7.1 and added the missing instructions part to the `README`.

As part of the PR I fixed issues related to using Dart SDK bundled in the Flutter distribution, we're now able to configure the `PUB_CACHE` variable in the very same way the [`PubCache`](https://github.com/SpineEventEngine/base/pull/589) is configured for the Dart proto plugin.

I have also updated the `base` and `web` versions and fixed related deprecations.

The issue is also related to the parent Gradle deps management issue https://github.com/SpineEventEngine/config/issues/156.

yuri-sergiichuk committed:
Add ability to use `pub` from a custom `PUB_CACHE` location as per the `PubCache` implementation in `base`.

yuri-sergiichuk committed:
Add some notes on setting up the dev env.

yuri-sergiichuk committed:
Update version.

yuri-sergiichuk committed:
Update `config`.

yuri-sergiichuk committed:
Update `.gitignore`.

yuri-sergiichuk committed:
Reformat as per the new config.

yuri-sergiichuk committed:
Upgradle v6.7.1

yuri-sergiichuk committed:
Update `gretty` version.

yuri-sergiichuk committed:
Fix usage of the deprecated API.

yuri-sergiichuk commented:
@dmdashenkov PTAL.

The build is failing because the [`web`](https://github.com/SpineEventEngine/web/pull/159) is not yet merged.

dmdashenkov commented on README.md:
````diff
@@ -1,3 +1,110 @@
+   ```bash
+   pub global activate protoc_plugin
+   ```
+
````
Please kill this emply line.

dmdashenkov reviewed (approved):
LGTM with a minor comment.

yuri-sergiichuk committed:
Drop extra empty line.

yuri-sergiichuk committed:
Make `PUB_CACHE` env optional.

The presence of the `PUB_CACHE` env is not a required condition while we may wanna try to rely on the default installation paths.
